astropy/table/info.py

```diff
@@ -0,0 +1,234 @@
+"""
+Table method and functions related to providing information about
+columns and tables.
+"""
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import sys
+import os
+
+import numpy as np
+
+from ..extern import six
+from ..utils import OrderedDict
+
+__all__ = ['column_info_factory']
+
+OPTIONS = ('meta', 'stats')
+
+def column_info_factory(names, funcs):
+    """
+    Factory to create a function that can be used as an ``option``
+    for outputting table or column summary information.
+
+    Examples
+    --------
+    >>> from astropy.table.info import column_info_factory
+    >>> from astropy.table import Column
+    >>> c = Column([4., 3., 2., 1.])
+    >>> mystats = column_info_factory(names=['min', 'median', 'max'],
+    ...                               funcs=[np.min, np.median, np.max])
+    >>> c.info(option=mystats)
+    min = 1.0
+    median = 2.5
+    max = 4.0
+
+    Parameters
+    ----------
+    names: list
+        List of information attribute names
+    funcs: list
```

Comment:
One could consider by default using a method on col with the same name, i.e., `func = lambda col: getattr(col, name)`. For `stats`, that would make some mixins work by default.  (but probably premature!)